lib/Conch/Controller/DatacenterRack.pm

```diff
@@ -149,6 +149,10 @@ sub update ($c) {
 		}
 	}
 
+	if($input->{role}) {
+		$input->{datacenter_rack_role_id} = delete $input->{role};
+	}
+
```

Comment:
better (and one line):

`$input->{datacenter_rack_role_id} = delete $input->{role} if exists $input->{role};`